### PR TITLE
Update all exported comments and start example_test.go for godoc

### PR DESCRIPTION
**Pull request by JohnRoesler**
This is explains godocs pretty helpfully: https://godoc.org/github.com/fluhus/godoc-tricks

What the examples look like:
image

### CODE REVIEW

File: job.go [hunk 1]
```diff
@@ -110,13 +101,21 @@ func (j *Job) periodDuration() (time.Duration, error) {
...
 func (j *Job) NextScheduledTime() time.Time {
 	return j.nextRun
 }
 
-// GetWeekday returns which day of the week the job will run on
-// This should only be used when .Weekday(...) was called on the job.
-func (j *Job) GetWeekday() time.Weekday {
-	return j.startDay
+// GetAt returns the specific time of day the Job will run at
+func (j *Job) GetAt() string {
```
Comment (arjunmahishi) [inline job.go]:
```suggestion
func (j *Job) GetScheduledTime() string {
```

File: job.go [hunk 2]
```diff
@@ -110,13 +101,21 @@ func (j *Job) periodDuration() (time.Duration, error) {
...
+// NextScheduledTime returns the time of the Job's next scheduled run
 func (j *Job) NextScheduledTime() time.Time {
 	return j.nextRun
 }
 
-// GetWeekday returns which day of the week the job will run on
-// This should only be used when .Weekday(...) was called on the job.
-func (j *Job) GetWeekday() time.Weekday {
-	return j.startDay
+// GetAt returns the specific time of day the Job will run at
```
Comment (arjunmahishi) [inline job.go]:
```suggestion
// GetScheduledTime returns the specific time of day the Job will run at
```

File: job.go (hunk 1 above)
Comment (arjunmahishi) [inline job.go]: Would returning `time.Time` instead of `string` be better here?

File: gocron.go [hunk 3]
```diff
@@ -20,16 +12,27 @@ package gocron
...
 	"runtime"
 	"strconv"
 	"strings"
 )
 
-// Locker provides a method to lock jobs from running
-// at the same time on multiple instances of gocron.
-// You can provide any locker implementation you wish.
+// Error declarations for gocron related errors
+var (
```
Comment (arjunmahishi) [inline gocron.go]: `ErrTimeFormat`, `ErrParamsNotAdapted`, `ErrNotAFunction`, `ErrPeriodNotSpecified` and `ErrParameterCannotBeNil` are unused

File: job.go (hunk 1 above)
Comment (JohnRoesler) [reply job.go]: I agree, time would be better. I'm going to leave that for #14

Comment (JohnRoesler) [reply job.go]: I like the rename, I'll update

**Comment by Streppel** [comment]
@arjunmahishi made some good points but I think this looks good now

**Comment by JohnRoesler** [comment]
Agreed - great comments - one will be addressed in another issue / PR

**Pull request merged**